stan/math/rev/core/vari.hpp

```diff
@@ -138,10 +171,23 @@ class vari {
    * See the discussion of "plugging the memory leak" in:
    *   http://www.parashift.com/c++-faq/memory-pools.html
    */
-  static inline void operator delete(void* /* ignore arg */) { /* no op */
+  static inline void operator delete(
+      void* /* ignore arg */) noexcept { /* no op */
   }
 };
 
+// For backwards compatability the default is double
+using vari = vari_value<double>;
+
+struct vari_printer {
+  std::ostream& o_;
+  int i_{0};
+  vari_printer(std::ostream& o, int i) : o_(o), i_(i) {}
+  template <typename T>
+  inline void operator()(T*& x) const {
+    o_ << i_ << "  " << x << "  " << x->val_ << " : " << x->adj_ << std::endl;
+  }
+};
```

Comment:
Is this just for debugging or you use that anywhere?

Reply:
I removed this when we switched back to the virtual case